### PR TITLE
Numeric Outcome DLC refactor

### PR DESCRIPTION
**Pull request by nkohen**
Re-worked CET language in numeric outcome specifications and added a diagram, as well as some general clean-up.

(Renames `CETCompression.md` -> `NumericOutcomeCompression.md`)

Related to #77.

### CODE REVIEW

File: NumericOutcome.md [hunk 1]
```diff
@@ -2,30 +2,31 @@
...
+When dealing with enumerated outcomes, DLCs require only a single nonce and Contract Execution
+Transactions (CETs) are claimed using a single oracle attestation.
+This scheme results in DLCs which contain a unique adaptor signature for every possible outcome, which
+is only feasible if the number of possible outcomes is of manageable size. 
 
 If an outcome can be any of a large range of numbers, then using a simple enumeration of
 all possible numbers in this range is unwieldy.
 We optimize for this case by using numeric decomposition in which the oracle signs each digit of the outcome
-individually so that many possible outcomes can be [compressed](CETCompression.md) into a single CET by ignoring certain digits.
+individually so that many possible outcomes can be [compressed](NumericOutcomeCompression.md) into a single CET adaptor signature by
```
Comment (Tibo-lg) [inline NumericOutcome.md]: Is it necessary to keep the CET here?

File: NumericOutcome.md [hunk 2]
```diff
@@ -88,12 +90,12 @@ If `begin_interval_1` is strictly greater than `0`, then the interval between `0
...
 * `begin_interval` MUST strictly increase.
 
-## Contract Execution Transaction Calculation
+## Contract Execution Transaction Calculation and Signing
 
 Given the offerrer's [payout function](PayoutCurve.md), a `total_collateral` amount and [rounding intervals](#rounding-intervals), we wish to compute a list of pairs
 of digit prefixes (i.e. arrays of integers) and Satoshi values.
-Each of these pairs will then be turned into a CET whose adaptor point is [computed from the digit prefix](CETCompression.md#adaptor-points-with-multiple-signatures) and whose
-output values will be equal to the Satoshi payout and `total_collateral` minus that payout.
+Each of these pairs will then be turned into a CET whose adaptor point used for signing is [computed from the digit prefix](CETCompression.md#adaptor-points-with-multiple-signatures) and
```
Comment (Tibo-lg) [inline NumericOutcome.md]: I think you need to update the link

File: NumericOutcome.md [hunk 3]
```diff
@@ -111,21 +113,31 @@ at the unmodified function's derivatives.
...
 For example, if you have two constant-valued intervals in a row with the same value, these must be merged.
 
-Finally, once these intervals have been computed, the [CET compression](CETCompression.md#cet-compression) algorithm is run on each constant-valued interval which generates
-a digit prefix (list of integers) to be paired with the (constant) payout for that interval.
-For variable-payout intervals, a unique CET is constructed for every `event_outcome` where all digits of that `event_outcome` are included
-in the array of integers and the Satoshi payout is equal to the output of the modified function for that `event_outcome`.
+Finally, once these intervals have been computed, the [numeric outcome compression](NumericOutcomeCompression.md#numeric-outcome-compression) algorithm is run on each constant-valued interval
+which generates a digit prefix (list of integers) to be paired with the (constant) payout for that interval.
+Only a single CET is required for each interval (as these intervals have constant payouts) where each CET receives multiple adaptor signatures.
```
Comment (Tibo-lg) [inline NumericOutcome.md]: Nit: to avoid repeating CET maybe: "where each CET" -> "and each of them"

File: NumericOutcomeCompression.md [hunk 4]
```diff
@@ -1,87 +1,88 @@
...
-computed as `s(1..m) = s1 + s2 + ... + sm` which can be used to broadcast the CET.
+When the oracle broadcasts its `n` attestations `s1, ..., sn`, the corresponding aggreate adaptor secret
+can be computed as `s(1..m) = s1 + s2 + ... + sm` which can be used to broadcast a corresponding CET.
 
 #### Rationale
 
-This design allows implementations to re-use all [transaction construction code](Transactions.md) without modification
-because every CET needs as input exactly one adaptor point just like in the single-nonce setting.
+This design allows implementations to re-use all [transaction construction](Transactions.md) and signing code without modification
+as every CET adaptor signature needs as input exactly one adaptor point just like in the single-nonce setting.
```
Comment (Tibo-lg) [inline NumericOutcomeCompression.md]: as before, is it necessary to keep CET (I don't mind that much tbh)

**Review by Tibo-lg — COMMENTED** [review]
Much more understandable! Just a couple of nits

**Review by Tibo-lg — APPROVED** [review]

**Review by benthecarman — APPROVED** [review]
LGTM

**Pull request merged**